### ISSUE TITLE
accuracy

### ISSUE THREAD

**Issue by nainadhingra2012** [ahmetgunduz/Real-time-GesRec#5]
Hi Ahmet,

I trained the classifier using Egogesture dataset. But the validation accuracy is just around 50% and also training accuracy is around 60%.

I am using ResNext101 architecture

Am I missing anything?

**Comment by ahmetgunduz** [comment]
Hi Naina, 

You should use pretrained models on Jester dataset and fine tune them on EgoGesture in order to get the accuracy level in the article. I am trying to upload the models but it is taking time because of some technical issues. In the meantime you can train your models on Jester first if you want to see the whole process. 

Besides that, usage of augmentation methods is critical for exploring the full potential of the model.

**Comment by ahmetgunduz** [comment]
I uploaded pretrained models. Please check **pretrained models** section of updated README file.

**Comment by nainadhingra2012** [comment]
Hi Ahmet,

Thank you for the models. I tried using those models, particularly, resnet-101 on egogesture.
I get a mismatch in the network. Did you use some other network for training?

Traceback (most recent call last):
  File "main.py", line 170, in <module>
    model.load_state_dict(checkpoint['state_dict'])
  RuntimeError: Error(s) in loading state_dict for DataParallel:
	size mismatch for module.conv1.weight: copying a param with shape torch.
Size([64, 1, 7, 7, 7]) from checkpoint, the shape in current model is torch.Size([64, 3, 3, 7, 7]).
	size mismatch for module.fc.weight: copying a param with shape torch.Size([83, 2048]) from checkpoint, the shape in current model is torch.Size([84, 2048]).
	size mismatch for module.fc.bias: copying a param with shape torch.Size([83]) from checkpoint, the shape in current model is torch.Size([84]).

I think the pretrained models are trained on a slightly different network.Could you share the network?

**Comment by ahmetgunduz** [comment]
Hi @nainadhingra2012 , you are right. 
Please comment these lines if you want to use resnet-101. Sorry that I forgot about this. 

```python
if opt.modality == 'RGB' and opt.model != 'c3d':
            print("[INFO]: RGB model is used for init model")
model = _modify_first_conv_layer(model,3,3) ##### Check models trained (3,7,7) or (7,7,7)
```

**Comment by nainadhingra2012** [comment]
Hey Ahmet, I commented those lines and also used the script:

#!/bin/bash
python main.py \
	--root_path /home/ndhingra/Real-time-GesRec/Real-time-GesRec \
	--video_path /media/storage/ndhingra/EgoGesture/images \
	--annotation_path annotation_EgoGesture/egogestureall.json \
	--result_path results \
	--resume_path /media/storage/ndhingra/pretrained_models/egogesture_resnext_101_Depth_32.pth \
	--dataset egogesture \
	--sample_duration 8 \
    --learning_rate 0.01 \
    --model resnetl \
	--model_depth 10 \
	--resnet_shortcut A \
	--batch_size 8 \
	--n_classes 83 \
	--n_finetune_classes 83 \
	--n_threads 16 \
	--checkpoint 1 \
	--modality Depth \
	--train_crop random \
	--n_val_samples 1 \
	--test_subset test \
     --n_epochs 100 \



I am getting error:

  File "main.py", line 167, in <module>
    assert opt.arch == checkpoint['arch']
AssertionError

and 


loading checkpoint /media/storage/ndhingra/pretrained_models/egogesture_resnext_101_Depth_32.pth
Traceback (most recent call last):
  File "main.py", line 170, in <module>
    model.load_state_dict(checkpoint['state_dict'])
  File "/home/ndhingra/Real-time-GesRec/pyt/lib/python3.5/site-packages/torch/nn/modules/module.py", line 769, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for DataParallel:
	Unexpected key(s) in state_dict: "module.layer1.0.downsample.0.weight", "module.layer1.0.downsample.1.weight", "module.layer1.0.downsample.1.bias", "module.layer1.0.downsample.1.running_mean", "module.layer1.0.downsample.1.running_var", "module.layer1.0.downsample.1.num_batches_tracked", "module.layer2.0.downsample.0.weight", "module.layer2.0.downsample.1.weight", "module.layer2.0.downsample.1.bias", "module.layer2.0.downsample.1.running_mean", "module.layer2.0.downsample.1.running_var", "module.layer2.0.downsample.1.num_batches_tracked", "module.layer3.0.downsample.0.weight", "module.layer3.0.downsample.1.weight", "module.layer3.0.downsample.1.bias", "module.layer3.0.downsample.1.running_mean", "module.layer3.0.downsample.1.running_var", "module.layer3.0.downsample.1.num_batches_tracked", "module.layer4.0.downsample.0.weight", "module.layer4.0.downsample.1.weight", "module.layer4.0.downsample.1.bias", "module.layer4.0.downsample.1.running_mean", "module.layer4.0.downsample.1.running_var", "module.layer4.0.downsample.1.num_batches_tracked".

**Comment by ahmetgunduz** [comment]
@nainadhingra2012 , You are trying to use ResNeXt-101 as a pretrained model but you have specified wrong parameters. 

> --sample_duration 8
--model resnetl
--model_depth 10
--resnet_shortcut A 

These should be as following :
`--sample_duration 32
--model resnext
--model_depth 101
--resnet_shortcut B 
`
please see my answer for another issue, in which I explain the pretrained models naming convention 
https://github.com/ahmetgunduz/Real-time-GesRec/issues/8#issuecomment-481617730

**Comment by nainadhingra2012** [comment]
Hi Ahmet,
    --model resnext
--model_depth 101
--resnet_shortcut B

I was using these settings as I was directly calling main.py. But, I am not sure if sample duration could e a problem. Thanks :) It is working now.

I tested your model with the above settings and I have test.json as a result of that test. I want to know which file out of all the files given on GitHub did you use to calculate accuracy and other measures for test data as mentioned in the paper.

In other words, how do I process test.json in terms of evaluating measures.
